### ISSUE TITLE
ODataRoute not working correctly

### ISSUE THREAD

**Issue by matthawley** [Microsoft/aspnet-api-versioning#39]
We have an explicit ODataRoute defined in our ODataController

`[ApiVersion("1.0"), ApiVersion("1.1")]`
`public class TestController : ODataController {`
`  [ODataRoute("Test(Name={name},Token={token})"), HttpGet]`
`  public IHttpActionResult Get([FromODataUri] string name, [FromODataUri] string token) {`
`    return Ok(new Test());`
`  }`
`}`

However, when we execute this endpoint it returns a 400 bad request saying it cannot find our specified route for the given api version. After poking around the code, the issue is that there's a type mismatch set on the EdmModel annotation value.

In src/Microsoft.AspNet.OData.Versioning/Web.OData/Routing/VersionedAttributeRoutingConvention.cs

`line 79: var apiVersion = Model.GetAnnotationValue<ApiVersion>(Model)`

Whereas in src/Microsoft.AspNet.OData.Versioning/Web.OData/Builder/VersionedODataModelBuilder.cs

`line 142: model.SetAnnotationValue( model, new ApiVersionAnnotation( apiVersion ) );`

Because `ApiVersionAnnotation` is being used instead of `ApiVersion` (or vice-versa), the controller is getting ignored from attribute routing.

**Comment by commonsensesoftware** [comment]
There were actually two places causing this problem, but you're completely right. Sadly, the unit test was written with the same incorrect annotation value, which produced a false positive. It took me several hours to track down the cause. Thanks so much for digging in and tracking this down. Ultimately the controller was correctly resolved, but the corresponding action was not; unless the key value segment was exactly **key** (e.g. the default route convention).  I documented this in Issue #40.

The changes can be seen in PR #41. The fix is in and I've published version [2.0.1](https://www.nuget.org/packages/Microsoft.AspNet.OData.Versioning/2.0.1) of the package. That should fix the issue. If you find that the update does not resolve the issue for you, we can setup a repro and start investigating.
